### DISCUSSION TITLE
How to build and push zstd image?

### DISCUSSION THREAD

**Discussion by slonopotamus** [containerd/nerdctl#3222] [Q&A] [answered]
Basically, subj. I do not see options to specify image compression format in neither `nerdctl build` nor `nerdctl push` as of `nerdctl version 2.0.0-rc.0`.

There is `nerdctl image convert --zstd --oci`, but it feels very inefficient (both from storage and time POV) to first produce gzip image and then convert it to zstd, especially in my context when images are tens of GBs in size.

Looking at other tools:
* Buildah has `buildah push --compression-format zstd`
* Podman has `podman push --compression-format zstd`
* kaniko has `/kaniko/executor --compression zstd`
* BuildKit has `buildctl --output type=image,oci-mediatypes=true,compression=zstd,push=true`

**Comment by AkihiroSuda** [comment, marked as answer]
```bash
nerdctl build --output type=image,oci-mediatypes=true,compression=zstd,push=true,name=example.com/foo .
```
